azure/Kqlmagic/kql_engine.py

```diff
@@ -151,7 +151,7 @@ def validate(self, **options):
     }
     _SECRET_KEYS = {ConnStrKeys.CLIENTSECRET, ConnStrKeys.APPKEY, ConnStrKeys.PASSWORD, ConnStrKeys.CERTIFICATE_THUMBPRINT}
     _NOT_INHERITABLE_KEYS = {ConnStrKeys.APPKEY, ConnStrKeys.ALIAS}
-    _OPTIONAL_KEYS = {ConnStrKeys.TENANT, ConnStrKeys.ALIAS}
+    _OPTIONAL_KEYS = {ConnStrKeys.TENANT, ConnStrKeys.ALIAS, ConnStrKeys.CLIENTID}
```

Comment:
I am not sure this is correct? as for some of the combinations clientId is not optional

Reply:
should I delete it entirely here?